### PR TITLE
Refactor template alignments to keep a fixed parcellation

### LINKED ISSUE #86: Keep a fixed parcellation when computing templates
`_align_images_to_template` instantiates a bunch of `PairwiseAlignment` estimators, each of which generate their own parcellation of the data. I would argue that a single parcellation should be computed once when instantiating the template for the following reasons:

1. Obvious speed gains;
2. The whole dataset could be used to derive more precise regions (especially for ward and kmeans), rather than working with pairs of subjects;
3. Better control when fitting the template.

Best,
PL

### PR DESCRIPTION
**Pull request by pbarbarant**
Closes #86. Uses the new backend to distribute template computations across parcels.

- [x] Docstrings
- [x] Unit tests
- [x] Adapt `examples/plot_template_alignment.py`

### CODE REVIEW

**Comment by pbarbarant** [comment]
I refactored the template example to close #100 . Template is now Procrustes-based.

**Comment by pbarbarant** [comment]
The doc build is now around 10min instead of over 30min, which is acceptable if we want to showcase examples with realistic use cases. @emdupre do you think we can close #92 ?

**Comment by emdupre** [comment]
Yes, that seems reasonable to me ! Thank you, @pbarbarant !

File: examples/plot_template_alignment.py [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
-In this tutorial, we show how to better predict new contrasts for a target
-subject using many source subjects corresponding contrasts. For this purpose,
-we create a template to which we align the target subject, using shared information.
-We then predict new images for the target and compare them to a baseline.
+In this tutorial, we show how to improve inter-subject similarity using a template
+computed across multiple source subjects. For this purpose, we create a template
+using Procrustes alignment (hyperalignment) to which we align the target subject,
+using shared information. We then compare the voxelwise similarity between the
+target subject and the template to the similarity between the target subject and
+the anatomical euclidean average of the source subjects.
```
Comment (bthirion) [inline examples/plot_template_alignment.py:11]:
```suggestion
the anatomical Euclidean average of the source subjects.
```

File: examples/plot_template_alignment.py [hunk 2]
```diff
@@ -64,22 +65,17 @@
 template_train = []
 for i in range(5):
     template_train.append(concat_imgs(imgs[i]))
-target_train = df[df.subject == "sub-07"][df.acquisition == "ap"].path.values
 
-# For subject sub-07, we split it in two folds:
-#   - target train: sub-07 AP contrasts, used to learn alignment to template
-#   - target test: sub-07 PA contrasts, used as a ground truth to score predictions
-# We make a single 4D Niimg from our list of 3D filenames
+# sub-07 will be our left-out subject.
```
Comment (bthirion) [inline examples/plot_template_alignment.py:69]:
```suggestion
# sub-07 (that is 5th in the list) will be our left-out subject.
```

**Review by bthirion — COMMENTED** [review]
LGT

**Pull request merged**